framework/src/base/FEProblem.C

```diff
@@ -3459,10 +3459,14 @@ FEProblem::computeResidual(NonlinearImplicitSystem &/*sys*/, const NumericVector
   {
     computeResidualType(soln, residual, _kernel_type);
   }
-  catch(MooseException & e)
-  {
-    mooseError("Impossible");
```

Comment:
I don't know really.  @permcody added it in 9ad3cd19e in January of this year.  I guess he thought it was impossible, but MOOSE makes the impossible... possible.